client/src/components/UserProfile.js

```diff
@@ -1,11 +1,63 @@
-import React from "react";
+import React, { useState } from "react";
+import Dropzone from "react-dropzone";
 
 const UserProfile = (props) => {
+  const [userInfoState, setUserInfoState] = useState(props.user);
+  const [imageData, setImageData] = useState({
+    image: {},
+  });
+  const handleImageUpload = (acceptedImage) => {
+    setImageData({
+      ...imageData,
+      image: acceptedImage[0],
+    });
+  };
+
+  const addProfileImage = async (event) => {
+    event.preventDefault();
+    const newImageBody = new FormData();
+    newImageBody.append("image", imageData.image);
+    try {
+      const response = await fetch(`/api/v1/users/${userInfoState.id}`, {
+        method: "PATCH",
+        headers: {
+          Accept: "image/jpeg",
+        },
+        body: newImageBody,
+      });
+      if (!response.ok) {
+        throw new Error(`${response.status} (${response.statusText})`);
+      }
+      const { body } = await response.json();
+      setUserInfoState({
+        ...userInfoState,
+        image: body.image,
+      });
+    } catch (error) {
+      console.error(`Error in addProfileImage Fetch: ${error.message}`);
+    }
+  };
   return (
     <>
-      <h1>{props.userData?.username}</h1>
-      <h3>{props.userData?.email}</h3>
-      <div className="profile-pic"></div>
+      <h1>{userInfoState?.username}</h1>
+      <h3>{userInfoState?.email}</h3>
+      <img className="profile-pic" src={userInfoState?.image}></img>
```

Comment:
not gonna hold it against you, but ideally we actually dont use the optional chaining operator here. While it works, it would be better if no img tag appears on the page at all if one doesnt exist. Instead you could do this (instead of an if..else)
```
{userInfoState.image ? <img className="profile-pic" src={userInfoState?.image}></img> : null}
```